.github/workflows/python-build-and-test.yaml

```diff
@@ -19,11 +22,11 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Install uv
-        uses: astral-sh/setup-uv@v3
+        uses: astral-sh/setup-uv@v4
         with:
           enable-cache: true
           cache-dependency-glob: pyproject.toml
-          cache-suffix: tox
+          cache-suffix: ${{ matrix.python-version }}-tox
```

Comment:
This is the cause of the caching issue identified in https://github.com/ccao-data/ccao/pull/33. Basically, `uv` [disables caching in CI by default](https://github.com/astral-sh/setup-uv?tab=readme-ov-file#disable-cache-pruning) because it's slower to recover the cache than it is to just `uv pip install` (wild).

It's _supposed to_ cache built-from-source wheels even when pruning is enabled. However, there's a race condition in the testing setup since the first-to-finish jobs are the ones to populate the cache, and those jobs won't have the build-from-source wheel.

The easy fix here is to just give each Python version its own cache.

Reply:
**[Praise]** This is wild 🤯  Great find!